### PR TITLE
Better initialization of S3 storage

### PR DESCRIPTION
**Pull request by excitoon**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Fixed bug with globs in S3 table function, region from URL was not applied to S3 client configuration.

### CODE REVIEW

File: src/IO/S3/PocoHTTPClient.h [hunk 1]
```diff
@@ -19,6 +19,8 @@ struct PocoHTTPClientConfiguration : public Aws::Client::ClientConfiguration
     const RemoteHostFilter & remote_host_filter;
 
     PocoHTTPClientConfiguration(const Aws::Client::ClientConfiguration & cfg, const RemoteHostFilter & remote_host_filter_);
+
+    void updateSchemeAndRegion();
```
Comment (alexey-milovidov) [inline src/IO/S3/PocoHTTPClient.h:23]: Missing comment.

**Comment by alexey-milovidov** [comment]
What is the scenario - is it when globs expand to different regions?
Is it possible to provide a test?

**Comment by alexey-milovidov** [comment]
The rest of the code LGTM.

**Comment by excitoon** [comment]
@alexey-milovidov It would be possible to add test if 1) we had access to some real S3 instance; or 2) I can make custom S3 server in Python which would check for signature and region but that's pretty complicated because I would need to have special name of host.

**Comment by tooptoop4** [comment]
@excitoon are IAM role supported for private buckets?

**Pull request merged**